test/vmtests/vmtests/imagecreator/test_imagecreator.py

```diff
@@ -89,6 +89,11 @@ def run_image_creator_test(
     base_ssh_config_path = IMAGECREATOR_TEST_CONFIGS_DIR.joinpath("ssh-base-config.yaml")
     customizer_config_path_obj = add_ssh_to_config(base_ssh_config_path, username, ssh_public_key, close_list)
 
+    # add previewFeatures section if distro is Fedora
+    if distro.lower() == "fedora":
```

Comment:
Follow `add_ssh_to_config`'s pattern and parse, modify and output the YAML file using a YAML library.